var/spack/repos/builtin/packages/qt-base/package.py

```diff
@@ -36,7 +36,7 @@ def get_list_url(qualname):
     # Default dependencies for all qt-* components
     generator("ninja")
     depends_on("cmake@3.16:", type="build")
-    depends_on("pkgconfig", type="build")
+    depends_on("pkgconfig", type="build", when=not bool(MACOS_VERSION))
```

Comment:
```suggestion
    depends_on("pkgconfig", type="build", when="platform=linux")
```

Instead of using `MACOS_VERSION`, I'd suggest explicitly adding the dependency when running on Linux.